homework-submission/react-hyf/react-hyf-client/src/App.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import './App.css';
+import Header from './components/Header';
+import TodoInput from './components/TodoInput';
+import TodoItem from './components/TodoItem';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [],
+      nextId: 5,
+    };
+
+    this.addTodo = this.addTodo.bind(this);
+    this.removeTodo = this.removeTodo.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchTodos().then(todos => {
+      this.setState({
+        todos,
+      });
+    });
+  }
+
+  fetchTodos = async () => {
+    const response = await fetch('/todos');
+    const data = await response.json();
+    return data;
+  };
+
+  addTodo(todoText) {
+    let todos = [...this.state.todos];
+    todos.push({ id: this.state.nextId, text: todoText });
+    this.setState({
+      todos: todos,
+      nextId: this.state.nextId + 1,
```

Comment:
As the previous Id comment, if you used an id generator the only place you will deal with it is here.